membership-attribute-service/app/services/AccountDetailsFromZuora.scala

```diff
@@ -110,7 +110,15 @@ class AccountDetailsFromZuora(
       nonGiftContactAndSubscriptions <- SimpleEitherT.rightT(nonGiftContactAndSubscriptionsFor(contact))
       contactAndSubscriptions <- checkForGiftSubscription(userId, nonGiftContactAndSubscriptions, contact)
       catalog <- SimpleEitherT.rightT(futureCatalog)
-      filtered = applyFilter(filter, contactAndSubscriptions, catalog)
+      subsWithRecognisedProducts = contactAndSubscriptions.filter(_.subscription.plan(catalog).product(catalog) match {
+        case _: Product.ContentSubscription => true
+        case Product.UnknownProduct => false
+        case Product.Membership => true
+        case Product.GuardianPatron => true
+        case Product.Contribution => true
+        case Product.Discounts => false
+      })
```

Comment:
this is the important bit, it filters out any that are unknown product or discounts as the primary product of the subscription